packages/web/src/components/SignUpForm/SignUpForm.tsx

```diff
@@ -8,112 +8,105 @@ import {
-
-  const [email, setEmail] = useState('');
-  const [emailError, setEmailError] = useState('');
-  const [emailTouched, setEmailTouched] = useState(false);
-
-  const [password, setPassword] = useState('');
-  const [passwordError, setPasswordError] = useState('');
-  const [passwordTouched, setPasswordTouched] = useState(false);
-
-  const [passwordMatch, setPasswordMatch] = useState('');
-  const [passwordMatchError, setPasswordMatchError] = useState('');
-  const [passwordMatchTouched, setPasswordMatchTouched] = useState(false);
-
-  const [isValid, setIsValid] = useState(false);
+  const initialValues = {
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+  };
 
-  // I've decided to keep it here for now, but later probably we can use 'validate' or/and 'Formik' npm libs.
-  const validateEmail = (emailToTest) => {
-    // eslint-disable-next-line
-    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-    return re.test(String(emailToTest).toLowerCase());
+  const emailErrorMessage = (email: string): string | null => {
+    if (!isEmail(email)) {
+      return 'Email is invalid.';
+    }
+    return null;
   };
 
-  useEffect(() => {
-    setIsValid(
-      !emailError 
-      && !passwordError 
-      && !passwordMatchError 
-      && !firstNameError 
-      && !lastNameError
-    );
-  }, [
-    emailError,
-    passwordError,
-    passwordMatchError,
-    firstNameError,
-    lastNameError,
-  ]);
-
-  // validation and error setting for First Name
-  useEffect(() => {
+  const firstNameErrorMessage = (firstName: string): string | null => {
     if (firstName.length < 2) {
-      setFirstNameError('First name has to be greater than 1 symbols.');
-    } else {
-      setFirstNameError('');
+      return 'First name has to be greater than 1 symbols.';
     }
-  }, [firstName]);
+    return null;
+  };
 
-  // validation and error setting for Last Name
-  useEffect(() => {
+  const lastNameErrorMessage = (lastName: string): string | null => {
     if (lastName.length < 2) {
-      setLastNameError('Last name has to be greater than 1 symbols.');
-    } else {
-      setLastNameError('');
+      return 'Last name has to be greater than 1 symbols.';
     }
-  }, [lastName]);
+    return null;
+  };
 
-  // validation and error setting for Email
-  useEffect(() => {
-    if (!validateEmail(email)) {
-      setEmailError('Email is invalid.');
-    } else {
-      setEmailError('');
+  const passwordErrorMessage = (password: string): string | null => {
+    if (password.length < 8) {
+      return 'Password has to be greater than 8 symbols.';
     }
-  }, [email]);
+    return null;
+  };
+
+  const {
+    values,
+    errors,
+    touched,
+    handleChange,
+    handleBlur,
+  }: {
+    values: any;
+    errors: any;
+    touched: any;
+    handleChange: any;
+    handleBlur: any;
```

Comment:
These could be something more specific, like `string[]`, `boolean`, `() => void`